modules/data-entry/src/main/frontend/src/questionnaire/Subject.jsx

```diff
@@ -223,45 +234,115 @@ function SubjectContainer(props) {
+  // This holds the full form JSON, once it is received from the server
+  let [ subject, setSubject ] = useState(null);
+  // Error message set when fetching the data from the server fails
+  let [ error, setError ] = useState();
+  // whether the subject has been deleted
+  let [ deleted, setDeleted ] = useState();
+
+  let globalLoginDisplay = useContext(GlobalLoginContext);
+
+  if (deleted) {
+    return null;
+  }
+
+  // Fetch the subject's data as JSON from the server.
+  // The data will contain the subject metadata,
+  // such as authorship and versioning information.
+  // Once the data arrives from the server, it will be stored in the `data` state variable.
+  let fetchSubjectData = () => {
+    fetchWithReLogin(globalLoginDisplay, `/Subjects/${id}.progeny.deep.json`)
+      .then((response) => response.ok ? response.json() : Promise.reject(response))
+      .then(handleSubjectResponse)
+      .catch(handleError);
+  };
+
+  // Callback method for the `fetchData` method, invoked when the data successfully arrived from the server.
+  let handleSubjectResponse = (json) => {
+    getSubject(json);
+    setSubject({data: json});
+  };
+
+  // Callback method for the `fetchData` method, invoked when the request failed.
+  let handleError = (response) => {
+    setError(response);
+    setSubject({});  // Prevent an infinite loop if data was not set
+  };
+
+  // Fetch this Subject's data
+  useEffect(() => {
+    fetchSubjectData();
+  }, [id]);
+
+  if (!subject?.data) {
+    return (
+      <Grid item><CircularProgress/></Grid>
+    );
+  }
+
+  if (error) {
+    return (
+      <Grid item>
+        <Typography variant="h2" color="error">
+          Error obtaining subject data: {error.status} {error.statusText ? error.statusText : error.toString()}
+        </Typography>
+      </Grid>
+    );
+  }
+
+  let identifier = subject?.data?.identifier ? subject.data.identifier : id;
+  let label = subject?.data?.type?.label || "Subject";
+  let title = `${label} ${identifier}`;
+  let path = subject?.data ? subject.data["@path"] : "/Subjects/" + id;
+  let action = <DeleteButton
+                 entryPath={path}
+                 entryName={title}
+                 entryType={label}
+                 shouldGoBack={true}
+                 onComplete={() => setDeleted(true)}
+                 buttonClass={classes.subjectHeaderButton}
+                 size={"large"}
+               />
+  let parentDetails = subject?.data && subject.data['parents'] && getHierarchy(subject.data['parents']);
 
   return (
-    <Grid item className={classes.subjectHeader}>
+    subject?.data && <Grid item className={classes.subjectHeader}>
       {parentDetails && <Typography variant="overline">{parentDetails}</Typography>}
       <Typography variant="h2">{title}{action}</Typography>
       {
-        data && data['jcr:createdBy'] && data['jcr:created'] ?
+        subject?.data && subject.data['jcr:createdBy'] && subject.data['jcr:created'] ?
```

Comment:
```suggestion
        subject?.data?.['jcr:createdBy'] && subject?.data?.['jcr:created'] &&
```
Along with removing the `: ""` on line 342

Reply:
In addition, is there a reason we check for both? I.e. is there a case where only one of the two exists?